### ISSUE TITLE
Result worse than 3dresnet50

### ISSUE THREAD

**Issue by Guocode** [Guocode/SlowFast-Networks#2]
dataset:something-somethingv2
slowfast only get 31% acc
3d-resnet-50 get 36% acc
under the same settings.
If anyone has do tests on any other dataset, you can publish your score here.

**Comment by nemonameless** [comment]
There are no links between the fast and slow branches in your code , but this may not be the main reason to the relatively bad results.
Do you think it's related to the number of GPU and pre-training? How about using pre-train model? maybe  you just overfit？

**Comment by SidHard** [comment]
I test this code on my dataset，and the result worse than I3D. I guess the main reason is no pre-train model.

**Comment by Guocode** [comment]
@SidHard I find it's essential to pretrain for 3DCNN on imagenet, is your test of I3D and slowfastnet both trained from scratch?

**Comment by 95xueqian** [comment]
I have trained [this model](https://github.com/RI-CH/SlowFastNetworks) on ucf101 dataset , the accuracy on test dataset is only 40%

**Comment by dmenig** [comment]
> @SidHard I find it's essential to pretrain for 3DCNN on imagenet, is your test of I3D and slowfastnet both trained from scratch?

The paper exhibits results with random init, so It should be possible to have the same results with no imagenet pretraining. That's in fact, to me one of the impressive things in it.

**Comment by hulianyuyy** [comment]
i just trained the SlowFast Model on UCF101 but i got ~ 55% accuracy on test split1. i think it's due to action complexities and video lengths. maybe another reason is  hyperparameters. we shoule use our own hyperparameters on each dataset.

**Comment by idealboy** [comment]
> I test this code on my dataset，and the result worse than I3D. I guess the main reason is no pre-train model.

Yes, I think it converge very slow, and it have a swing about the acc and loss.

It is much worse than I3D with the same train datase and val dataset, and I have the same sampling strategy in the clip like my I3D training.

**Comment by dmenig** [comment]
I didn't have the ressources to test with the exact same parameters as the paper, but I reached 63% on kinetics in validation. However it does train from scratch which is very impressive. I do think it's because of my parameters because with the same parameters, I3D performs on par with this. But like @idealboy  it's way worse than other models on my private dataset.

**Comment by salmedina** [comment]
> i just trained the SlowFast Model on UCF101 but i got ~ 55% accuracy on test split1. i think it's due to action complexities and video lengths. maybe another reason is hyperparameters. we shoule use our own hyperparameters on each dataset.

Hello~ Which set of hyperparameters did you use for this? I was able to achieve 0.51 accuracy on UCF-101 by using the default parameters described on the paper. Thank you!

**Comment by hulianyuyy** [comment]
> > i just trained the SlowFast Model on UCF101 but i got ~ 55% accuracy on test split1. i think it's due to action complexities and video lengths. maybe another reason is hyperparameters. we shoule use our own hyperparameters on each dataset.
> 
> Hello~ Which set of hyperparameters did you use for this? I was able to achieve 0.51 accuracy on UCF-101 by using the default parameters described on the paper. Thank you!

i used batchsize 16, lr 0.01 and epoch 200 when trained with Pytorch. I think this network needs pretrained model to help it perform well otherwise it will have a bad performance.

**Comment by salmedina** [comment]
> > > i just trained the SlowFast Model on UCF101 but i got ~ 55% accuracy on test split1. i think it's due to action complexities and video lengths. maybe another reason is hyperparameters. we shoule use our own hyperparameters on each dataset.
> > 
> > 
> > Hello~ Which set of hyperparameters did you use for this? I was able to achieve 0.51 accuracy on UCF-101 by using the default parameters described on the paper. Thank you!
> 
> i used batchsize 16, lr 0.01 and epoch 200 when trained with Pytorch. I think this network needs pretrained model to help it perform well otherwise it will have a bad performance.

All the way to 200 epochs? I ran it for only 80 epochs.  Which optimizer did you use? Do you have any idea if anyone has released a pretrained model on Kinetics? I was thinking about pretraining it on Kinetics and release it if no one else has done it. But it is going to take me weeks...

**Comment by hulianyuyy** [comment]
> > > > i just trained the SlowFast Model on UCF101 but i got ~ 55% accuracy on test split1. i think it's due to action complexities and video lengths. maybe another reason is hyperparameters. we shoule use our own hyperparameters on each dataset.
> > > 
> > > 
> > > Hello~ Which set of hyperparameters did you use for this? I was able to achieve 0.51 accuracy on UCF-101 by using the default parameters described on the paper. Thank you!
> > 
> > 
> > i used batchsize 16, lr 0.01 and epoch 200 when trained with Pytorch. I think this network needs pretrained model to help it perform well otherwise it will have a bad performance.
> 
> All the way to 200 epochs? I ran it for only 80 epochs. Which optimizer did you use? Do you have any idea if anyone has released a pretrained model on Kinetics? I was thinking about pretraining it on Kinetics and release it if no one else has done it. But it is going to take me weeks...

emmm i just used 200 epochs because i tried it when training on with other modes and it performs well. This model could bring out better if equipped with pretrained model but due to limits of GPUs
i can't train it with Kinetics...Not having a pretrained model is a pity to it.

**Comment by hulianyuyy** [comment]
> > > > i just trained the SlowFast Model on UCF101 but i got ~ 55% accuracy on test split1. i think it's due to action complexities and video lengths. maybe another reason is hyperparameters. we shoule use our own hyperparameters on each dataset.
> > > 
> > > 
> > > Hello~ Which set of hyperparameters did you use for this? I was able to achieve 0.51 accuracy on UCF-101 by using the default parameters described on the paper. Thank you!
> > 
> > 
> > i used batchsize 16, lr 0.01 and epoch 200 when trained with Pytorch. I think this network needs pretrained model to help it perform well otherwise it will have a bad performance.
> 
> All the way to 200 epochs? I ran it for only 80 epochs. Which optimizer did you use? Do you have any idea if anyone has released a pretrained model on Kinetics? I was thinking about pretraining it on Kinetics and release it if no one else has done it. But it is going to take me weeks...

oh i just forgot to tell i used standard SGD to optimize.
